src/app/clusters/software_diagnostics_server/software_diagnostics_server.cpp

```diff
@@ -16,11 +16,83 @@
  */
 
 #include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterface.h>
 #include <app/CommandHandler.h>
+#include <app/MessageDef/AttributeDataElement.h>
 #include <app/util/af.h>
+#include <app/util/attribute-storage.h>
+#include <lib/core/Optional.h>
+#include <platform/PlatformManager.h>
 
 using namespace chip;
+using namespace chip::app;
 using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::SoftwareDiagnostics::Attributes;
+using chip::DeviceLayer::PlatformManager;
+
+namespace {
+
+class SoftwareDiagosticsAttrAccess : public AttributeAccessInterface
+{
+public:
+    // Register for the SoftwareDiagnostics cluster on all endpoints.
+    SoftwareDiagosticsAttrAccess() : AttributeAccessInterface(Optional<EndpointId>::Missing(), SoftwareDiagnostics::Id) {}
+
+    CHIP_ERROR Read(ClusterInfo & aClusterInfo, TLV::TLVWriter * aWriter, bool * aDataRead) override;
+
+private:
+    CHIP_ERROR ReadIfSupported(CHIP_ERROR (PlatformManager::*getter)(uint64_t &), TLV::TLVWriter * aWriter);
+};
+
+SoftwareDiagosticsAttrAccess gAttrAccess;
+
+CHIP_ERROR SoftwareDiagosticsAttrAccess::Read(ClusterInfo & aClusterInfo, TLV::TLVWriter * aWriter, bool * aDataRead)
+{
+    if (aClusterInfo.mClusterId != SoftwareDiagnostics::Id)
+    {
+        // We shouldn't have been called at all.
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    *aDataRead = true;
```

Comment:
Should it be set to true even if a given feature is unsupported?

Reply:
I asked the same question to Boris when he introduced this new API. Here is his reply
"dataRead is not signaling success or failure. It's signaling whether this was an attribute the interceptor thought it's supposed to handle or not."

With this new API, this part of logic is platform and example independent. If the feature is not supported on a specific platform, this will be handled by platform layer and return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE